### DISCUSSION TITLE
How i can add EntityGraph to BlazePersist

### DISCUSSION THREAD

**Discussion by MishaBucha** [Blazebit/blaze-persistence#1884] [Q&A]
Hi all, I am facing a problem, I don't understand how to add EntityGraph to my query for fetching. Can you tell me if this is even supported in BlazePersist or if it's not possible? Code:
```
 public List<DOMAIN> findAll(String... attribute) {
        BlazeCriteriaBuilder cb = BlazeCriteria.get(criteriaBuilderFactory);
        BlazeCriteriaQuery<DOMAIN> query = cb.createQuery(getDomainClass());
        BlazeRoot<DOMAIN> root = query.from(getDomainClass());

        EntityGraph<DOMAIN> entityGraph = entityManager.createEntityGraph(getDomainClass());
        entityGraph.addAttributeNodes(attribute);

        CriteriaBuilder<DOMAIN> criteriaBuilder = query.createCriteriaBuilder(entityManager);
        return criteriaBuilder.getResultList();
    }
```

**Comment by beikov** [comment]
Hi there. You can create a JPA `Query` with the `CriteriaBuilder.getQuery()` method. On that, you can apply the entity graph via `setHint` like usually.

**Reply by curso007** [reply]
Hi, I tried that and I got an exception "not yet implemented"

**Reply by jwgmeligmeyling** [reply]
Does your query use advanced query features like CTE's or set functions? Afaik those have indeed a few more limitations
